### PR TITLE
test: fix TestParseFilerelayLogUpdatedOrNewCreated

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 
part of #1997 

### What is changed and how it works?

The reason the previous test sometimes worked was that the ctx was quickly cancelled and `relayLogUpdatedOrNewCreated` hadn't had time to open the meta file

so i extended the time of ctx, and prepare the meta file inadvance

### Check List 

Tests 

 - Unit test

Code changes

Side effects

 - Possible performance regression

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/streamer/reader_test.go [hunk 1]
```diff
@@ -289,8 +289,9 @@ func (t *testReaderSuite) TestParseFilerelayLogUpdatedOrNewCreated(c *C) {
 	}
 
 	// no valid update for relay sub dir, timeout, no error
-	ctx1, cancel1 := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	ctx1, cancel1 := context.WithTimeout(context.Background(), time.Second)
 	defer cancel1()
+	t.createMetaFile(c, relayDir, filename, uint32(offset), notUsedGTIDSetStr)
```
Comment (lance6716) [inline pkg/streamer/reader_test.go:294]: createMetaFile should be put together with `// create the current relay log file and write some events`

**Review by lance6716 — COMMENTED** [review]
the function of this test should be renamed to `TestParseFileRelayLogUpdatedOrNewCreated` (capitalize R)

rest lgtm

**Comment by Ehco1996** [comment]
seems test : DM-DropAddColumn optimistic still not steable 🤣 
https://ci.pingcap.net/blue/organizations/jenkins/dm_ghpr_new_test/detail/dm_ghpr_new_test/1981/pipeline

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
> seems test : DM-DropAddColumn optimistic still not steable 🤣
> https://ci.pingcap.net/blue/organizations/jenkins/dm_ghpr_new_test/detail/dm_ghpr_new_test/1981/pipeline

these failures are recorded in https://github.com/pingcap/dm/issues/1997

**Review by okJiang — APPROVED** [review]

**Pull request merged**